### PR TITLE
Tooltips lab Phase 1 Integration

### PR DESCRIPTION
**Pull request by ChesterSng**
This PR will integrate three functionalities into `Create Tooltip`

1. `Callout` will be created (along with `triggershape` if there is no `selection`)
2. `Callout` has animation triggered by `triggershape`
3. `Auto resizing textbox` added to `Callout`

- [x] Integrate resizable textbox 

_Note: This PR also merges latest `dev-release` into `tooltips-lab`_

### CODE REVIEW

File: PowerPointLabs/PowerPointLabs/ActionFramework/TooltipsLab/AssignTooltip/AssignTooltipEnabledHandler.cs [hunk 1]
```diff
@@ -14,7 +14,14 @@ class AssignTooltipEnabledHandler : EnabledHandler
         protected override bool GetEnabled(string ribbonId)
         {
             Selection currentSelection = this.GetCurrentSelection();
-            return ShapeUtil.IsSelectionMultipleSameShapeType(currentSelection);
+            if (!ShapeUtil.IsSelectionShape(currentSelection))
```
Comment (YuPeiHenry) [inline PowerPointLabs/PowerPointLabs/ActionFramework/TooltipsLab/AssignTooltip/AssignTooltipEnabledHandler.cs]: Is there a reason why `ShapeUtil.IsSelectionShape(currentSelection)` is not returned directly?

File: PowerPointLabs/PowerPointLabs/ActionFramework/TooltipsLab/CreateTooltip/CreateTooltipEnabledHandler.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace PowerPointLabs.ActionFramework.TooltipsLab
+{
+    [ExportEnabledRibbonId(TooltipsLabText.CreateTooltipTag)]
+    class CreateTooltipEnabledHandler : EnabledHandler
+    {
+        protected override bool GetEnabled(string ribbonId)
+        {
+            PowerPointSlide currentSlide = this.GetCurrentSlide();
+            if (currentSlide != null)
```
Comment (YuPeiHenry) [inline PowerPointLabs/PowerPointLabs/ActionFramework/TooltipsLab/CreateTooltip/CreateTooltipEnabledHandler.cs]: Is there a reason for not returning `currentSlide != nil` directly?

File: PowerPointLabs/PowerPointLabs/ThisAddIn.cs [hunk 3]
```diff
@@ -937,12 +937,15 @@ private void ThisAddInSlideSelectionChanged(PowerPoint.SlideRange sldRange)
 
                 UpdateRecorderPane(sldRange.Count, slideID);
                 TimerLab.TimerLab.IsTimerEnabled = true;
+                PictureSlidesLab.PictureSlidesLab.IsPictureSlidesEnabled = true;
```
Comment (YuPeiHenry) [inline PowerPointLabs/PowerPointLabs/ThisAddIn.cs]: Is PictureSlideLab related to Tooltips lab? I'm didn't know about this.

File: PowerPointLabs/PowerPointLabs/ThisAddIn.cs [hunk 4]
```diff
@@ -937,12 +937,15 @@ private void ThisAddInSlideSelectionChanged(PowerPoint.SlideRange sldRange)
...
                 TimerLab.TimerLab.IsTimerEnabled = true;
+                PictureSlidesLab.PictureSlidesLab.IsPictureSlidesEnabled = true;
             }
             else
             {
                 UpdateRecorderPane(sldRange.Count, -1);
                 TimerLab.TimerLab.IsTimerEnabled = false;
                 UpdateTimerPane(false);
+                PictureSlidesLab.PictureSlidesLab.IsPictureSlidesEnabled = false;
+                ShutDownPictureSlidesLab();
```
Comment (YuPeiHenry) [inline PowerPointLabs/PowerPointLabs/ThisAddIn.cs]: Same question on PicturesSlidesLab.

File: PowerPointLabs/PowerPointLabs/ThisAddIn.cs [hunk 5]
```diff
@@ -975,6 +978,8 @@ private void ThisAddInSlideSelectionChanged(PowerPoint.SlideRange sldRange)
                     prev = presentation.Slides[slideIndex - 1];
                 }
             }
+            Ribbon.RefreshRibbonControl("CreateTooltipButton");
+            Ribbon.RefreshRibbonControl("PictureSlidesLabButton");
```
Comment (YuPeiHenry) [inline PowerPointLabs/PowerPointLabs/ThisAddIn.cs]: Same question on PictureSlidesLab.

File: PowerPointLabs/PowerPointLabs/ActionFramework/TooltipsLab/CreateTooltip/CreateTooltipEnabledHandler.cs (hunk 2 above)
Comment (ChesterSng) [reply PowerPointLabs/PowerPointLabs/ActionFramework/TooltipsLab/CreateTooltip/CreateTooltipEnabledHandler.cs]: thats true then less lines need, i'll make the changes!

**Pull request merged**